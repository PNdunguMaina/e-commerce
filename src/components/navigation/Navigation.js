import { NavLink } from 'react-router-dom';

const Navigation = () => {
  const links = [
    { id: 1, path: '/', text: 'Home' },
    { id: 2, path: '/details', text: 'Details' },
  ];
  return (
    <div>
      <div className="logo-container">Air pollution</div>
      <nav>
        {links.map((link) => (
          <NavLink key={link.id} to={link.path}>
            {link.text}
          </NavLink>
        ))}
      </nav>
    </div>
  );
};

export default Navigation;
