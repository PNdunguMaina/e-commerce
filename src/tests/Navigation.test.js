import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Navigation from '../components/navigation/Navigation';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Navigation />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
