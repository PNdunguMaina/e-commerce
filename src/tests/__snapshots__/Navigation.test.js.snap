// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark sticky-top"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="NYX logo"
        className="d-inline-block align-top"
        height="30"
        src="logo.png"
        width="30"
      />
      NYX
    </span>
    <form
      className="d-flex"
    >
      <input
        aria-label="Search"
        className="me-2 form-control"
        onClick={[Function]}
        placeholder="Search..."
        type="search"
      />
      <button
        className="btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</nav>
`;
